### PR TITLE
fix(drop): polly retry execute loop

### PR DESCRIPTION
**Pull request by alexirion10**
SMO & SqlClient upgrade, 
KillDatabase wrapped in retry loop

this is after more indepth testing with Angelo

### CODE REVIEW

File: src/SmoFacade/Database.cs [hunk 1]
```diff
@@ -95,6 +97,12 @@ public List<BackupMetadata> RecentBackups()
       return backups;
     }
 
+    public void SingleUserMode()
```
Comment (Kaneraz) [inline src/SmoFacade/Database.cs:100]: Good addition, but not used?

**Review by Kaneraz — APPROVED** [review]
Looks like more aggressive retries, is the hope here to have be able to catch it on the retry before something else connects?

I know we made some timeout changes to support the longer running backups, do we need to make a similar change to drop so we can handle rollbacks of large (10min+) transactions?

File: src/SmoFacade/Database.cs (hunk 1 above)
Comment (alexirion10) [reply src/SmoFacade/Database.cs:100]: correct

**Comment by alexirion10** [comment]
@Kaneraz - I was thinking over some of the math on the retries. Yes more aggressive, but since this is exponential the top end is very reasonable:
before (ms):
1000^1, 1000^2, 1000^3, 1000^4, 1000^5, 1000^6,
1000, 10000, 100000, 1000000, ...., .... ms
1, 10, 1000 (16.6667min), 10000 (166.66 min), 100000 (27.7 hours) seconds

after (ms):
10^1, 10^2, 10^3, 10^4, 10^5, 10^6
10, 100, 1000, 10000, 100000, 1000000 ms
.01, .1, 1, 10, 100 (1.66 min) , 1000 (16.6667 min) seconds

**Comment by alexirion10** [comment]
also I added the retry logic to `Drop` in this PR. 
see `policy.Execute(() => { _database.Parent.KillDatabase(_database.Name); });`

**Comment by Kaneraz** [comment]
Yep, 27 hours is a bit absurd

**Pull request merged**